### PR TITLE
 Add test to check dnf command alias in the help output

### PR DESCRIPTION
**Pull request by pkratoch**
Some packages needed to be added to the dockerfiles in order to have
the yum/yum4 command available.

DNF base command aliases on various systems:
 - Fedora <= 30: "yum" from package "dnf-yum"
 - Fedora >= 31: "yum" from package "yum"
 - RHEL 8: "yum" from package "yum"
 - RHEL 7 (EPEL): "yum4" from package "nextgen-yum4"

Additionally, on all of these systems, there is package "dnf" providing
command "dnf" and "dnf-2" or "dnf-3" based on python version ("dnf-2" in
RHEL7, "dnf-3" otherwise)

### CODE REVIEW

**Comment by pkratoch** [comment]
The tag centos__eq__7 doesn't work for some reason despite commit https://github.com/rpm-software-management/ci-dnf-stack/commit/d30622545cc307789e5aa8b8a62d5253150c8b27#diff-fb5dc39ddb7e42f372ef4f2513af02e8

And btw, nextgen-yum4 is currently broken in the dnf-nightly repo (missing commit https://github.com/rpm-software-management/dnf/commit/660eec1969fc0679d7febfb3a2cb26f9cf7678ec ), but it can be built using the TYPE=distro

**Comment by pkratoch** [comment]
> The tag centos__eq__7 doesn't work for some reason despite commit [d306225#diff-fb5dc39ddb7e42f372ef4f2513af02e8](https://github.com/rpm-software-management/ci-dnf-stack/commit/d30622545cc307789e5aa8b8a62d5253150c8b27#diff-fb5dc39ddb7e42f372ef4f2513af02e8)

The problem was (and still is) that tags are evaluated with logical OR, meaning that when there are e.g. tags ``@not.with_os=rhel__eq__7`` and ``@not.with_os=centos__eq__7``, the scenario is not skipped because one of the tags evaluates to true.

The tags used in this test should be working now. I replaced ``@not.with_os`` with complementary ``@use.with_os``.

File: Dockerfile.f31 [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+os=$OSVERSION" > /opt/behave/behave.ini
+
+# build test repos from sources
+RUN set -x && \
+    cd /opt/behave/fixtures/specs/ && \
+    ./build.sh --force-rebuild
+
+
+VOLUME ["/opt/behave/junit"]
+WORKDIR /opt/behave
```
Comment (lukash) [inline Dockerfile.f31]: Could the new Dockerfile go into a separate commit?

File: dnf-behave-tests/features/help.feature [hunk 2]
```diff
@@ -26,3 +26,51 @@ Scenario: Command help
...
+    And stdout contains "usage: <command>"
+
+Examples:
+        | args      | command  |
+        | help      | dnf      |
+        | --help    | dnf      |
+        | help      | dnf-3    |
+        | --help    | dnf-3    |
+        | help      | yum      |
+        | --help    | yum      |
```
Comment (lukash) [inline dnf-behave-tests/features/help.feature]: Maybe flip the two columns for a bit better readability?

File: Dockerfile.f31 (hunk 1 above)
Comment (pkratoch) [reply Dockerfile.f31]: Done.

File: dnf-behave-tests/features/help.feature (hunk 2 above)
Comment (pkratoch) [reply dnf-behave-tests/features/help.feature]: Done.

File: dnf-behave-tests/features/steps/cmd.py [hunk 3]
```diff
@@ -7,7 +7,12 @@
 import re
 import sys
 from datetime import datetime
-from itertools import zip_longest
+
+PY3 = sys.version_info.major >= 3
+if PY3:
+    from itertools import zip_longest
+else:
+    from itertools import izip_longest as zip_longest
```
[the remark this answers is not in the document]
Comment (pkratoch) [reply dnf-behave-tests/features/steps/cmd.py]: I don't see much difference between the two and the same is used in ``steps.fixtures.httpd``, so I'd rather stay consistent.

[the remark this answers is not in the document]
Comment (lukash) [reply dnf-behave-tests/features/steps/cmd.py]: Okay.
